File: javascript/recurcive_binary_search.js
function recurcive_binary_search(A,p,r,x) {
	if (p>r)
    return "Not found";
  if (p<=r){
   var q=(p+r)/2
   q=Math.floor(q)
    if (A[q]==x)
      return q;
    if (A[q]>x)
      return recurcive_binary_search(A,p,q-1,x);
    if (A[q]<x)
      return recurcive_binary_search(A,q+1,r,x);
  }
}


var a = recurcive_binary_search([1, 2, 3333],0, [1, 2, 3].length-1, 3333);
alert(a)

File: python/recurcive_binary_search.py
def recurcive_binary_search(A,p,r,x):
  if p>r:
    return "Not found"
  elif p<=r:
    q=(p+r)//2
    if A[q]==x:
      return q
    elif A[q]>x:
      return recurcive_binary_search(A,p,q-1,x)
    elif A[q]<x:
      return recurcive_binary_search(A,q+1,r,x)
  
  
A=[3,5,7]
a=recurcive_binary_search(A,0,(len(A)-1),78)
print(a)